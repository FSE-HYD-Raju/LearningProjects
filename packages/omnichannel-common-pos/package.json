{
	"name": "omnichannel-common-pos",
	"version": "111.2.0",
	"description": "Common actions, stores and components for omnichannel",
	"main": "lib/index",
	"types": "lib/index.d.ts",
	"oc:es6": "src/index",
	"scripts": {
		"test": "../../testSinglePackage.sh $(pwd)",
		"test-watch": "jest --watch",
		"lint": "eslint tests src --max-warnings=0 ",
		"test-ci": "jest",
		"lint-fix": "eslint src tests --fix",
		"flow": "flow; test $? -eq 0 -o $? -eq 2",
		"clean": "rm -rf lib",
		"build-ts": "tsc --pretty",
		"copy-assets": "copyfiles --up 1 'src/**/*.@(scss|jpg|jpeg|gif|png|svg)' ./lib",
		"build": "cross-env NODE_ENV=production BABEL_ENV=production BABEL_DISABLE_CACHE=1 babel -s inline -d lib/ src/ && npm run build-ts && npm run copy-assets && rimraf -rf messages",
		"watch": "cross-env NODE_ENV=production BABEL_ENV=production BABEL_DISABLE_CACHE=1 babel -w -s inline -d lib/ src/ && rimraf -rf messages",
		"prepublish": "npm run clean && npm run build"
	},
	"jest": {
		"testRegex": "(/tests/.*)\\.test\\.js$",
		"coverageDirectory": "jestcoverage",
		"coverageReporters": [
			"lcov",
			"html"
		],
		"setupFiles": [
			"../../jest-setup.js"
		]
	},
	"author": "",
	"license": "ISC",
	"publishConfig": {
		"registry": "https://artifactory.qvantel.net/artifactory/api/npm/npm-local"
	},
	"dependencies": {
		"alt": "0.18.6",
		"alt-utils": "2.0.0",
		"autosuggest-highlight": "2.1.1",
		"axios": "0.18.0",
		"babel-cli": "6.26.0",
		"babel-core": "6.26.0",
		"babel-eslint": "7.2.3",
		"babel-jest": "23.6.0",
		"babel-loader": "7.1.3",
		"babel-plugin-add-module-exports": "0.2.1",
		"babel-plugin-module-resolver": "3.1.0",
		"babel-plugin-react-intl": "2.4.0",
		"babel-plugin-react-transform": "2.0.2",
		"babel-plugin-transform-class-properties": "6.24.1",
		"babel-plugin-transform-decorators-legacy": "1.3.4",
		"babel-plugin-transform-es2015-classes": "6.24.1",
		"babel-plugin-transform-proto-to-assign": "6.26.0",
		"babel-plugin-transform-react-constant-elements": "6.23.0",
		"babel-plugin-transform-react-inline-elements": "6.22.0",
		"babel-plugin-transform-react-pure-class-to-function": "1.0.1",
		"babel-plugin-transform-react-remove-prop-types": "0.4.20",
		"babel-plugin-transform-runtime": "6.23.0",
		"babel-plugin-typecheck": "3.9.0",
		"babel-polyfill": "6.26.0",
		"babel-preset-es2015": "6.24.1",
		"babel-preset-es2015-loose": "7.0.0",
		"babel-preset-react": "6.24.1",
		"babel-preset-stage-0": "6.24.1",
		"babel-register": "6.26.0",
		"babel-runtime": "6.26.0",
		"bootstrap-sass": "3.3.7",
		"byte-size": "4.0.4",
		"classnames": "2.2.6",
		"connected-react-router": "4.3.0",
		"copyfiles": "2.0.0",
		"country-data": "0.0.31",
		"cross-env": "5.1.4",
		"crypto-js": "3.1.9-1",
		"css-ns": "1.2.1",
		"debug": "4.1.0",
		"diff": "3.4.0",
		"history": "4.7.2",
		"jwt-decode": "2.2.0",
		"langs": "2.0.0",
		"lodash": "4.17.11",
		"moment": "2.21.0",
		"moment-duration-format": "2.2.2",
		"object-assign": "4.1.1",
		"omnichannel-flow-pos": "111.2.0",
		"path": "0.12.7",
		"path-to-regexp": "2.3.0",
		"prop-types": "15.6.1",
		"ramda": "0.24.1",
		"rc-progress": "2.2.5",
		"rc-slider": "8.6.1",
		"react": "16.10.2",
		"react-autosuggest": "9.3.4",
		"react-bootstrap": "0.32.1",
		"react-collapse": "4.0.3",
		"react-confirm": "0.1.14",
		"react-dates": "20.1.0",
		"react-dom": "16.10.2",
		"react-dropzone": "3.13.4",
		"react-flatpickr": "3.6.4",
		"react-formal": "0.25.4",
		"react-formal-inputs": "2.1.1",
		"react-helmet": "5.2.0",
		"react-highlight-words": "0.11.0",
		"react-input-range": "1.3.0",
		"react-intl": "2.7.0",
		"react-motion": "0.5.2",
		"react-motion-ui-pack": "0.10.3",
		"react-onclickoutside": "6.7.1",
		"react-redux": "5.0.7",
		"react-router": "4.2.0",
		"react-router-dom": "4.2.2",
		"react-select": "1.2.1",
		"react-toastr": "3.0.0",
		"react-widgets": "3.5.0",
		"recompose": "0.30.0",
		"redux": "4.0.0",
		"redux-saga": "0.16.0",
		"redux-saga-tester": "1.0.379",
		"shallowequal": "0.2.2",
		"twix": "1.2.1",
		"universal-cookie": "2.1.2",
		"uuid": "3.2.1",
		"weasel-styles": "2.3.1",
		"yup": "0.21.3"
	},
	"devDependencies": {
		"@types/alt": "0.16.35",
		"@types/classnames": "2.2.6",
		"@types/jest": "23.3.9",
		"@types/lodash": "4.14.118",
		"@types/moment-duration-format": "2.2.2",
		"@types/node": "8.0.0",
		"@types/rc-slider": "8.2.3",
		"@types/react": "16.7.18",
		"@types/react-bootstrap": "0.32.14",
		"@types/react-collapse": "4.0.2",
		"@types/react-confirm": "0.1.2",
		"@types/react-dates": "17.1.5",
		"@types/react-dom": "16.0.5",
		"@types/react-flatpickr": "3.7.0",
		"@types/react-helmet": "5.0.7",
		"@types/react-highlight-words": "0.12.0",
		"@types/react-intl": "2.3.9",
		"@types/react-motion": "0.0.27",
		"@types/react-motion-ui-pack": "0.10.1",
		"@types/react-onclickoutside": "6.0.4",
		"@types/react-redux": "6.0.0",
		"@types/react-router": "4.0.25",
		"@types/react-select": "1.2.0",
		"@types/react-toastr": "3.0.3",
		"@types/recompose": "0.26.4",
		"@types/redux-actions": "2.3.0",
		"@types/redux-devtools": "3.0.44",
		"@types/redux-devtools-dock-monitor": "1.1.32",
		"@types/redux-devtools-log-monitor": "1.0.33",
		"@types/redux-logger": "3.0.6",
		"@types/seamless-immutable": "7.1.2",
		"@types/universal-cookie": "2.2.0",
		"@types/webpack": "4.4.0",
		"@types/webpack-env": "1.13.6",
		"@types/yup": "0.26.0",
		"babel-plugin-__coverage__": "11.0.0",
		"babel-plugin-transform-flow-strip-types": "6.22.0",
		"babel-plugin-transform-runtime": "6.23.0",
		"enzyme": "3.7.0",
		"enzyme-adapter-react-16": "1.10.0",
		"enzyme-to-json": "3.3.3",
		"eslint": "4.10.0",
		"eslint-config-omnichannel": "2.2.4",
		"eslint-config-prettier": "1.7.0",
		"eslint-config-react-app": "2.1.0",
		"eslint-loader": "1.9.0",
		"eslint-plugin-babel": "4.1.2",
		"eslint-plugin-flowtype": "2.39.1",
		"eslint-plugin-import": "2.8.0",
		"eslint-plugin-jsx-a11y": "5.1.1",
		"eslint-plugin-prettier": "2.6.0",
		"eslint-plugin-react": "7.4.0",
		"eslint-plugin-react-intl": "1.1.1",
		"flow-bin": "0.70.0",
		"jest": "23.6.0",
		"jest-junit": "3.6.0",
		"lint-staged": "4.3.0",
		"moxios": "0.4.0",
		"pre-commit": "1.2.2",
		"prettier": "1.6.1",
		"react-router-enzyme-context": "1.2.0",
		"react-test-renderer": "16.7.0",
		"redux-devtools": "3.4.1",
		"redux-devtools-dock-monitor": "1.1.3",
		"redux-devtools-log-monitor": "1.4.0",
		"redux-logger": "3.0.6",
		"rimraf": "2.6.2",
		"sinon": "4.3.0",
		"ts-jest": "23.10.5",
		"ts-loader": "5.3.1",
		"typescript": "3.1.6",
		"webpack": "4.26.1"
	},
	"commitInfo": "omni-ui-common-pos|184f176|1571059113|Venkata_Gangadhar_Kodela|develop"
}
