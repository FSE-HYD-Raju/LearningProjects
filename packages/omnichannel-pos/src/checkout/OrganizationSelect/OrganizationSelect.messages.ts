/* tslint:disable:max-line-length */
import { defineMessages, FormattedMessage } from "react-intl";

// Interface has been generated automatically. Please do not edit manually.
// For changes run in console: npm run rebuild-messages-interfaces
interface OrganizationSelectMessagesType {
	checkboxLabel: FormattedMessage.MessageDescriptor;
}
const OrganizationSelectMessages: OrganizationSelectMessagesType = defineMessages({
	checkboxLabel: {
		id: "OrganizationSelect-use-organization-checkbox-label",
		description: "Use organization select checkbox label",
		defaultMessage: "Legal Rep of Company"
	},
});

export default OrganizationSelectMessages;
export { OrganizationSelectMessages, OrganizationSelectMessagesType };
