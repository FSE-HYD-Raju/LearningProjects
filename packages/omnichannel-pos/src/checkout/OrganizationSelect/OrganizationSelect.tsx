import * as React from "react";
import { get } from "lodash";
import { Component } from "react";
import { AxiosResponse } from "axios";
import { FormalCheckbox, ErrorContainer, OrganizationResponse, ContextType, FormattedMessage } from "omnichannel-common-pos";
import messages from "./OrganizationSelect.messages";

export interface OrganizationSelectProps {
	organizations: Array<OrganizationFromApi>;
	showOrganizationSelect: boolean;
	selectedOrganizationId: string;
	onOrganizationSelectChange(organizationId: string): void;
	onOrganizationCheckboxChange(organizationId: boolean): void;
	getOrganizationById(id: string): Promise<AxiosResponse<OrganizationResponse>> | ErrorContainer;
}

export interface OrganizationSelectState {
	organizations: Array<OrganizationInfo>;
}

export interface OrganizationFromApi {
	relatedPersonId: string;
}

export interface OrganizationInfo {
	id: string;
	name: string;
}

/**
 * Display organization selection, if a user belongs to any organizations
 * so a user can perform checkout as organization.
 */
export default class OrganizationSelect extends Component<OrganizationSelectProps, OrganizationSelectState> {
	constructor(props: OrganizationSelectProps, context: ContextType) {
		super(props, context);

		this.state = {
			organizations: []
		};
	}

	componentWillMount() {
		this.fetchOrganizationNames();
	}

	/**
	 * Fetches organization names if needed
	 */
	fetchOrganizationNames = async () => {
		const { organizations, getOrganizationById } = this.props;

		if (organizations) {
			const fetches: Array<any> = [];
			organizations.forEach(organization => {
				fetches.push(getOrganizationById(organization.relatedPersonId));
			});

			const results = await Promise.all(fetches);

			if (results) {
				const organizations = results.map(result => {
					return {
						id: get(result, "data.id"),
						name: get(result, "data.attributes.formattedName")
					};
				});

				const selectedOrganizationId: string = get(organizations, "[0].id", null);
				this.props.onOrganizationSelectChange(selectedOrganizationId);

				this.setState({
					organizations
				});
			}
		}
	};

	render() {
		const { organizations } = this.state;

		if (!organizations || organizations.length === 0) {
			return null;
		}

		return (
			<React.Fragment>
				<FormalCheckbox
					id="use-organization-checkbox"
					checked={this.props.showOrganizationSelect}
					label={<FormattedMessage {...messages.checkboxLabel}/>}
					onChange={(value: boolean) => {
						this.props.onOrganizationCheckboxChange(value);
					}}
				/>

				{this.props.showOrganizationSelect && (
					<select
						className="custom-select"
						id="organization-select"
						value={this.props.selectedOrganizationId}
						onChange={event => {
							this.props.onOrganizationSelectChange(
								event.target.value
							);
						}}
					>
						{organizations.map((organization, key) => {
							return (
								<option
									key={key}
									value={organization.id}
									id={`organization-select-option-${
										organization.id
										}`}
								>
									{organization.name}
								</option>
							);
						})}
					</select>
				)}
			</React.Fragment>
		);
	}
}
