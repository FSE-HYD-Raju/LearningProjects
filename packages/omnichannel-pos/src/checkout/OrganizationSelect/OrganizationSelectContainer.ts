import * as React from "react";
import { connect } from "react-redux";
import {
	default as OrganizationSelect,
	OrganizationSelectProps
} from "./OrganizationSelect";
import { get } from "lodash";
import { AppState, OrganizationService } from "omnichannel-common-pos";

export interface OrganizationSelectContainerProps
	extends OrganizationSelectProps {
	CustomerCaseStore: any;
}

const mapStateToProps = (state: AppState, ownProps: OrganizationSelectContainerProps): OrganizationSelectProps => {
	const { CustomerCaseStore, ...rest } = ownProps;

	return {
		organizations: get(
			CustomerCaseStore,
			"activeCustomerCase.attributes.activeCustomer.personRelationships",
			[]
		),
		getOrganizationById: OrganizationService.getById,
		...rest
	};
};

export default connect(mapStateToProps)(OrganizationSelect);
