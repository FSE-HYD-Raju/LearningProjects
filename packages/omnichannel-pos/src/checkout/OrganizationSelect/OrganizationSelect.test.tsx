import * as React from "react";
import OrganizationSelect, { OrganizationSelectProps } from "./OrganizationSelect";
import { mountWithContext, shallowWithContext } from "omnichannel-common-pos";
import { ReactWrapper, ShallowWrapper } from "enzyme";

describe("OrganizationSelect", () => {
	let wrapper: ShallowWrapper | ReactWrapper;
	let props: OrganizationSelectProps;
	const onOrganizationSelectChange = jest.fn();
	const onOrganizationSelectCheckboxChange = jest.fn();
	beforeEach(() => {
		props = {
			organizations: [
				{
					relatedPersonId: "123"
				}
			],
			showOrganizationSelect: false,
			selectedOrganizationId: "",
			getOrganizationById: jest.fn(),
			onOrganizationSelectChange: onOrganizationSelectChange,
			onOrganizationCheckboxChange: onOrganizationSelectCheckboxChange
		};
	});

	afterEach(() => {
		if (wrapper) {
			wrapper.unmount();
		}
	});

	it("succeeds at shallow mount with min props", () => {
		wrapper = shallowWithContext(<OrganizationSelect {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("should show organization select dropdown when show organizationSelect checkbox is clicked", () => {
		props.showOrganizationSelect = true;
		const wrapper = mountWithContext(<OrganizationSelect {...props} />);
		wrapper.setState({ organizations: [{ id: "1", formattedName: "1. Company" }, { id: "2", formattedName: "2. Company" }] });
		wrapper.update();
		expect(onOrganizationSelectChange.mock.calls.length).toBe(1);
		const select = wrapper.find("#organization-select").at(0);
		select.simulate("change", {target: { value : "1"}});
		wrapper.update();
		expect(onOrganizationSelectChange.mock.calls.length).toBe(2);
	});
});
