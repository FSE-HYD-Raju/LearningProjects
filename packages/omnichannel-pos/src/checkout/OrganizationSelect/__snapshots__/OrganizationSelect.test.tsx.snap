// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationSelect succeeds at shallow mount with min props 1`] = `""`;
